samples/nrf9160/aws_fota/src/main.c

```diff
@@ -403,27 +403,28 @@ void main(void)
 
 	err = nrf_modem_lib_get_init_ret();
 	switch (err) {
+	case 0:
+		/* Initialization successful, no action required. */
+		break;
 	case MODEM_DFU_RESULT_OK:
 		printk("Modem firmware update successful!\n");
 		printk("Modem will run the new firmware after reboot\n");
-		k_thread_suspend(k_current_get());
-		break;
+		sys_reboot(SYS_REBOOT_COLD);
```

Comment:
I'm not sure whether the printk message above will be transmitted completely. Can you check and add k_sleep() if needed.

Reply:
The last line wasn't printed. Added a small delay, thanks!